source/javascripts/sliders2.js

```diff
@@ -158,6 +158,67 @@ tableApp2.controller('TableCtrl', function ($scope) {
 
     $scope.allLanguages = [
         {
+            name: "Safe Rust",
+            softname: "Up-and-coming systems lang",
+            markupName: "safe-rust",
+            comment: "//",
+
+            missingEnum: {
+                enforced: "yes",
+                desc: "Compiler enforced."
+            },
+            recursionStackOverflow: {
+                enforced: "no",
+                desc: "LLVM can compile tail calls in optimized mode, but this is not guaranteed, and, in fact, may silently go away if the compiler inserts a destructor call."
+            },
+            nullList: {
+                enforced: "yes",
+                desc: "Null pointers are never allowed, and Option must be explicitly handled."
+            },
+            deadLocks: {
+                enforced: "no",
+                desc: "Deadlocks are not considered unsafe."
+            },
+            missingMethodOrField: {
+                enforced: "yes",
+                desc: "Compiler Enforced."
+            },
+            wrongVaribleType: {
```

Comment:
In the table, you call this "Prevent Variable Reuse for Different Type" however, rust does allow shadowing. That is, you can say:

```rust
let x = "test";
let x = true;
```

But can't say:

```rust
let mut x = "test";
x = true;
```